File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        map<char,int> char_map;
        
        if(s.length()!=t.length()){//cant be anagram if unequal length
            return false;
        }
        for(char ch : s ){
            char_map[ch]++;
        }
        for(char ch : t ){
            char_map[ch]--;
        }

        for(const auto& pair : char_map){
            if(pair.second>0){
                return false;
            }
        }
    

        return true;
    }
};

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        smap=defaultdict(int)
        tmap=defaultdict(int)
        for c in s:
            smap[c]+=1
        for c in t:
            tmap[c]+=1

        if len(smap.items())!=len(tmap.items()):
            return False
        for k,v in smap.items():
            if tmap[k]!=v:
                return False
        return True